test/subspace-test-service/src/lib.rs

```diff
@@ -449,13 +451,22 @@ impl MockConsensusNode {
         &mut self,
     ) -> (
         Slot,
-        Option<OpaqueBundle<NumberFor<Block>, Hash, DomainHeader, Balance>>,
+        OpaqueBundle<NumberFor<Block>, Hash, DomainHeader, Balance>,
     ) {
-        let slot = self.produce_slot();
-
-        let bundle = self.notify_new_slot_and_wait_for_bundle(slot).await;
+        for _ in 0..MAX_PRODUCE_BUNDLE_TRY {
+            let slot = self.produce_slot();
+            if let Some(bundle) = self.notify_new_slot_and_wait_for_bundle(slot).await {
+                return (slot, bundle);
+            }
+        }
+        panic!("Failed to produce bundle after {MAX_PRODUCE_BUNDLE_TRY:?} tries, something must be wrong");
+    }
 
-        (slot, bundle)
+    /// Produce a new slot and wait for the slot is handled but not ensure bundle is submitted
+    pub async fn produce_slot_and_wait_for_handling(&mut self) -> Slot {
```

Comment:
I understand why the function exists but I dont think this should be used by anyone except MockConsensusNode to produce a block without bundle. 

We also make an assumption that bundle is always produced for tests. If we introduce this functionality, then I think we may end seeing edgecases. IMHO, we keep retrying and then either fail or suceed to produce bundle

Reply:
Not really, there are usage in the test for consensus block without bundle and we only assuming bundle is always produced when `produce_slot_and_wait_for_bundle_submission` or `produce_blocks_with_bundles` is used explicitly in the test.

Reply:
I disagree. If there are usages where bundle is not required, then it should simply just call produce a new consensus block. There is absolutely no need to call the this `produce_slot_and_wait_for_handling` instead. Also when i read it i'm not sure what exactly the function is handling here 🤔